packages/block-editor/src/components/url-input/index.js

```diff
@@ -84,12 +84,13 @@ class URLInput extends Component {
 			}, 100 );
 		}
 
-		// Only attempt an update on suggestions if the input value has actually changed.
 		if (
 			prevProps.value !== value &&
-			this.shouldShowInitialSuggestions()
+			! this.props.disableSuggestions &&
+			! this.isUpdatingSuggestions &&
+			! this.props.__experimentalShowInitialSuggestions
```

Comment:
Could you explain this change and why it's necessary? This change removes documentation in terms of:

1. The code comment explaining what's happening.
2. A well named function call which self-documents the check logic.

Reply:
The intention here was to bypass `shouldShowInitialSuggestions`, which was returning false because of `__experimentalShowInitialSuggestions`, and just copy its logic regarding the display of suggestions. 
I see now that there was a misunderstanding from my side regarding what initial suggestions meant so the `! this.props.__experimentalShowInitialSuggestions` part of my change does not make sense.

I am not sure how to proceed further here as I don't understand why `__experimentalShowInitialSuggestions` is set to false inside shouldShowInitialSuggestions
https://github.com/WordPress/gutenberg/blob/bd514b86b0c7866f4c92f0a395e57f660f0f7d22/packages/block-editor/src/components/url-input/index.js#L116

Reply:
> I am not sure how to proceed further here as I don't understand why __experimentalShowInitialSuggestions is set to false inside shouldShowInitialSuggestions

As far as I can see it isn't necessarily being set to `false` here. Rather `false` is used the _default_ value if `__experimentalShowInitialSuggestions` is _not_ provided as a prop (ie: is `undefined`). See [MDN for more details](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#default_values).

I still don't quite understand why you need to bypass the `this.shouldShowInitialSuggestions()` logic here? What was it about that which was breaking the fix you've tried to introduce here?

Reply:
Hi @getdave,
I am sorry for the late reply.

> What was it about that which was breaking the fix you've tried to introduce here?

You can see in the video below the behavior if I don't bypass `this.shouldShowInitialSuggestions()`:
when I click on the second link I see the suggestions from the first link. 
https://www.loom.com/share/79dedfab3b6942458d2b6fa6d2c9e34b

Let's look  at what `this.shouldShowInitialSuggestions()` returns: 
```
return 
(
! this.isUpdatingSuggestions && // suggestions aren't loaded
__experimentalShowInitialSuggestions && 
! ( value && value.length ) && // there is no value in the input
! ( suggestions && suggestions.length ) // there are no suggestions loaded
)
```
Please note that this would return true only when the user selects a text and clicks Add link.

In our case, when clicking the second link, `this.shouldShowInitialSuggestions()` always returns false because:
- there is a value in the input, namely the 2nd link
- there are suggestions available, from the 1st link.

What happens is that when I open the 1st link, the suggestions are updated [with the onFocus callback](https://github.com/WordPress/gutenberg/blob/bd514b86b0c7866f4c92f0a395e57f660f0f7d22/packages/block-editor/src/components/url-input/index.js#L226-L241). When clicking on the 2nd link, onFocus is no longer called and  the suggestions aren't updated in [componentDidUpdate](https://github.com/WordPress/gutenberg/blob/bd514b86b0c7866f4c92f0a395e57f660f0f7d22/packages/block-editor/src/components/url-input/index.js#L88-L93) because of `this.shouldShowInitialSuggestions`